pkg/config/expand.go

```diff
@@ -412,6 +412,59 @@ type varContext struct {
 	blueprint  Blueprint
 }
 
+/*
+A module reference is made by the use keyword and is subject to IGC constraints
+of references (ordering, explicitness). It has the following fields:
+  - ID: a module ID
+  - GroupID: the deployment group in which the module is *expected* to be found
+  - SourceGroupID: the deployment group from which the reference is made
+  - ExplicitInterGroup: a boolean value indicating whether the user made a
+    reference that superficially appears to be intergroup (i.e. they used
+    an explicit GroupID that differs from the context's GroupID)
+*/
+type modReference struct {
```

Comment:
I gave this thought since they are so close. I decided against since I don't think there is a "is a" relationship here in the first place. It might work from a purely counting lines of code perspective, but I don't think it facilitates much else since I don't think there would be much shared implementations.

Reply:
Thanks for the explanation.